### ISSUE TITLE
taxi which cannot reach origin of a trip triggers error

### ISSUE THREAD

**Issue by behrisch** [eclipse/sumo#9111]
there may be others which can serve the trip so this should be at most a warning

**Comment by namdre** [comment]
taxi simulation in networks which are not fully connected is not recommended (#8570)

**Comment by behrisch** [comment]
I think full connectedness is too hard a requirement. It should suffice that at any time a trip from A to B is requested we choose all taxis which can serve P->A->B->P where P is the current position. Even the random circling problem could be solved by just choosing a random destination D such that P->D->P is possible.

**Comment by namdre** [comment]
I agree in principle but the problems typically don't come from two disconnected graphs which are well connected individually but rather from lots of dead-ends at the network fringe.

**Comment by behrisch** [comment]
Yes, but I thought it might be an easier (or at least alternative) strategy to avoid those dead ends up front instead of fixing the net.

**Comment by namdre** [comment]
This may work for randomCircling put forces you to reject customers from those edges as well. Hence you might as well eliminate the edge from your simulation which leaves you with a fully connected network.

**Comment by behrisch** [comment]
No, the problem is, that we are not talking about a network which serves only taxis. There may be edges which are only used by trains or pedestrians. I think we should rather force the pedestrians to walk to the next junction if the taxi cannot turn around.

**Comment by namdre** [comment]
I think edges which are only for trains/pedestrians are tangential to this since pedestrians will walk to the next taxi-permitting edge anyway (when intermodal routing is in use). Forbidding taxis on some edges that would normally allow them but are actually dead-ends would indeed be a way to work around the problem (again, by restricting taxis to one/multiple fully connected graphs).

**Comment by behrisch** [comment]
Well at least the theoretic complexity is linear time for finding strongly connected components https://en.wikipedia.org/wiki/Tarjan%27s_strongly_connected_components_algorithm

**Comment by behrisch** [comment]
may also enable restricted service areas for different fleets
